lib/datasource/docker/index.ts

```diff
@@ -10,8 +10,8 @@ import type { HostRule } from '../../types';
 import { ExternalHostError } from '../../types/errors/external-host-error';
 import * as packageCache from '../../util/cache/package';
 import * as hostRules from '../../util/host-rules';
-import type { Http, HttpOptions, HttpResponse } from '../../util/http';
-import { HttpError } from '../../util/http/types';
+import type { Http } from '../../util/http';
+import { HttpError, HttpOptions, HttpResponse } from '../../util/http/types';
```

Comment:
I moved:
https://github.com/renovatebot/renovate/blob/15c393f79115c0a5fe6429396c5a9e51127ffc67/lib/util/http/types.ts#L27
to index.ts and changed the import.

Reply:
after this change, there are a lot of tests failed, so I revert it and I'm debugging it locally right now.
At this point, just one test fails now I think after this change the behavior has been changed.
![image](https://user-images.githubusercontent.com/98313426/155129574-545b8f6d-cb39-4447-8c4d-9251f4e58e47.png)

Reply:
@viceice
after moving HttpError from `types.ts` to `index.ts` I got circular dependency problem the solution was to remove legacy import from `index.ts`,
https://github.com/renovatebot/renovate/blob/93c2f7b1c025b23644178a9210fe99919387cc45/lib/util/http/index.ts#L24
 in this case the test above failed because there are some undefined properties like" `statuscode`, `body`, `url`..."
https://github.com/renovatebot/renovate/blob/93c2f7b1c025b23644178a9210fe99919387cc45/lib/util/http/legacy.ts#L7-L11
 and these properties are found in legacy.ts, so when running the tests without legacy import `statuscode` property is not initialized.
- is this problem related to the third part of this issue https://github.com/renovatebot/renovate/issues/11450? 
- should I delete legacy.ts ? or move what it contains and then delete the file itself?

thanks.

Reply:
Remove legacy is another later part. 

Let's move the error to index and export from there. Then add a separate import from got to legacy without re-export.

This should avoid the circular references.